### PR TITLE
fix: normalise `account_id = ''` to `account_id: undefined`

### PR DESCRIPTION
**Pull request by threepointone**
Disable whitespace for this review!

--- 

In older templates, (i.e made for wrangler 1.x), `account_id =''` is considered as a valid input, but then ignored. With wrangler 2, when running wrangler dev, we log an error, but it fixes itself after we get an account id. Much like https://github.com/cloudflare/wrangler2/issues/1329, the fix here is to normalise that value when we see it, and replace it with `undefined` while logging a warning.

This fix also tweaks the messaging for a blank route value to suggest some user action.

### CODE REVIEW

File: packages/wrangler/src/config/validation.ts [hunk 1]
```diff
@@ -670,20 +670,39 @@ function isValidRouteValue(item: unknown): boolean {
...
+
 /**
  * Normalize empty string to `undefined` by mutating rawEnv.route value.
  * As part of backward compatibility with Wrangler1 converting empty string to `undefined`
  */
 function mutateEmptyStringRouteValue(
-	rawEnv: RawEnvironment,
-	diagnostics: Diagnostics
+	diagnostics: Diagnostics,
+	rawEnv: RawEnvironment
```
Comment (JacobMGEvans) [inline packages/wrangler/src/config/validation.ts:698]: This a convention?

File: packages/wrangler/src/config/validation.ts [hunk 2]
```diff
@@ -670,20 +670,39 @@ function isValidRouteValue(item: unknown): boolean {
...
 ): RawEnvironment {
 	if (rawEnv["route"] === "") {
 		diagnostics.warnings.push(
-			`Route assignment ${JSON.stringify(rawEnv["route"])} will be ignored.`
+			`The "route" field in your configuration is an empty string and will be ignored.\n` +
+				`Please remove the "route" field from your configuration.`
```
Comment (JacobMGEvans) [inline packages/wrangler/src/config/validation.ts:703]: Allowing the empty string, then asking them to remove the field seems ambivalent.

File: packages/wrangler/src/config/validation.ts (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/config/validation.ts:698]: I see that it's common we are passing diagnostics first.

**Review by JacobMGEvans — APPROVED** [review]
We should have further discussions internally pertaining to `null` vs. `undefined` vs values like`[ ]` 
Cc: @petebacondarwin 

This isn't blocking and my question for messaging is also non-blocking

File: packages/wrangler/src/config/validation.ts (hunk 2 above)
Comment (threepointone) [reply packages/wrangler/src/config/validation.ts:703]: It's more actionable than just saying it'll be ignored, this gives them an action to take to remove the warning. 
It'll happen when they use an older project or template. We'll have a go at cleaning up the templates when we work on `generate`, but we should also give actionable advice.

**Pull request merged**